src/components/wasteCalendar/WasteReminderSettings.tsx

```diff
@@ -0,0 +1,361 @@
+    async (street: string) => {
+      setLoading(true);
+      setError(false);
+      const storedSettings = await getReminderSettings();
+
+      if (!areValidReminderSettings(storedSettings)) {
+        setError(true);
+        setLoading(false);
+      } else {
+        const newSettings: ReminderSettings = parseReminderSettings(storedSettings, street);
+
+        // TODO: fix label for stored settings of inactive types ans show them as well
+        // initialize uninitialized types with false
+        Object.keys(types).forEach((key) => {
+          newSettings.activeTypes[key] ??= { active: false };
+        });
+
+        dispatch({ type: ReminderSettingsActionType.OVERWRITE, payload: newSettings });
+        setLocalSelectedTime(newSettings.reminderTime);
+
+        setLoading(false);
+      }
+    },
+    [dispatch, setError, setLoading, types]
+  );
+
+  const onAbortIOS = useCallback(() => {
+    setShowDatePicker(false);
+    setLocalSelectedTime(state.reminderTime);
+  }, [state.reminderTime, setLocalSelectedTime, setShowDatePicker]);
+
+  const onAcceptIOS = useCallback(() => {
+    setShowDatePicker(false);
+    dispatch({ type: ReminderSettingsActionType.UPDATE_TIME, payload: localSelectedTime });
+  }, [localSelectedTime, setShowDatePicker, dispatch]);
+
+  const onDatePickerChange = useCallback(
+    (_, newTime?: Date) => {
+      if (device.platform === 'android') {
+        setShowDatePicker(false);
+
+        if (newTime) {
+          newTime.setMilliseconds(0);
+          newTime.setSeconds(0);
+          dispatch({ type: ReminderSettingsActionType.UPDATE_TIME, payload: newTime });
+        }
+      }
+
+      setLocalSelectedTime((time) => newTime || time);
+    },
+    [setLocalSelectedTime]
+  );
+
+  const onPressDayBefore = useCallback(
+    () => dispatch({ type: ReminderSettingsActionType.UPDATE_ON_DAY_BEFORE, payload: true }),
+    [dispatch]
+  );
+
+  const onPressDayOfCollection = useCallback(
+    () => dispatch({ type: ReminderSettingsActionType.UPDATE_ON_DAY_BEFORE, payload: false }),
+    [dispatch]
+  );
+
+  const updateSettings = useCallback(async () => {
+    const newState = { ...state };
+    let errorOccured = false;
+
+    await Promise.all(
+      Object.keys(state.activeTypes).map(async (typeKey) => {
+        const entry = state.activeTypes[typeKey];
+        // delete inactive entries
+        if (!entry.active && entry.storeId) {
+          const success = await deleteReminderSetting(entry.storeId);
+
+          errorOccured ||= !success;
+        }
+
+        if (entry.active) {
+          // update setting
+          const id = await updateReminderSettings({
+            ...locationData,
+            onDayBefore: state.onDayBefore,
+            reminderTime: `${state.reminderTime.getHours()}:${state.reminderTime.getMinutes()}`,
+            wasteType: typeKey
+          });
+
+          // save new id
+          newState.activeTypes[typeKey].storeId = id;
```

Comment:
can `newState.activeTypes` be null or undefined, so that `[typeKey]` would break?